sysl2/sysl/commands/runner.go

```diff
@@ -0,0 +1,73 @@
+package commands
+
+import (
+	"strings"
+
+	sysl "github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/parse"
+	"github.com/anz-bank/sysl/sysl2/sysl/syslutil"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type Runner struct {
+	commands []Command
+
+	Root   string
+	module string
+}
+
+func (r *Runner) Run(which string, fs afero.Fs, logger *logrus.Logger) error {
+	for _, cmd := range r.commands {
```

Comment:
This loop can be completely avoided, but if the execution logic is housed inside the command implementation created above.

Reply:
Design Suggestion: You could create implementations of the command interface for all the sysl commands and then map them using actions in the kingpin framework.

Reply:
As we discussed, I disagree that moving it to a kingpin.Action() would work and code duplication is not a good thing.

Switched to storing a map to avoid extra loops